setup.py

```diff
@@ -6,6 +6,31 @@
 import shlex
 import subprocess
 import sys
+from time import sleep
+
+
+def is_virtualenv():
+    return sys.base_prefix != sys.prefix
+
+
+if platform.system() == "Darwin":
+    major_version = int(platform.mac_ver()[0].split(".")[0])
+    if major_version < 12:
+        print(
+            "\033[1;91mWarning!\033[0m You are using an EOL, unsupported, and out-of-date OS."
+        )
+        sleep(3)
+
+# Don't show message when using our project tooling.
+if not is_virtualenv() or os.getenv("_PYAV_ACTIVATED", "") != "1":
+    print(
+        "\n\033[1;91mWarning!\033[0m You are installing from source.\n"
```

Comment:
is this of stuff really necessary? Have you been getting questions from users that are accidentally installing from source?

Reply:
related... https://github.com/PyAV-Org/PyAV/pull/1778